### ISSUE TITLE
Exception with unknown phone_number_type when accessing a candidate list

### ISSUE THREAD

**Issue by stuart-alatus** [merge-api/merge-java-client#41]
When accessing a candidate list from Postman, 
GET https://api.merge.dev/api/ats/v1/candidates?created_after=1953-11-23T13:06:29.055Z
I receive a full candidate list.  However when accessing through the merge-java-client using the same API call I get an exception.
```
Exception in thread "main" java.lang.NoSuchFieldError: READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE
	at com.fasterxml.jackson.databind.deser.std.EnumDeserializer.createContextual(EnumDeserializer.java:211)
	at com.fasterxml.jackson.databind.DeserializationContext.handleSecondaryContextualization(DeserializationContext.java:867)
	at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:659)
	at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:4956)
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4537)
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4475)
	at com.merge.api.resources.ats.types.PhoneNumberPhoneNumberType$Deserializer.deserialize(PhoneNumberPhoneNumberType.java:84)
	at com.merge.api.resources.ats.types.PhoneNumberPhoneNumberType$Deserializer.deserialize(PhoneNumberPhoneNumberType.java:75)
	at com.fasterxml.jackson.databind.deser.std.ReferenceTypeDeserializer.deserialize(ReferenceTypeDeserializer.java:204)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:158)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.vanillaDeserialize(BuilderBasedDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:217)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:359)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28)
	at com.fasterxml.jackson.databind.deser.std.ReferenceTypeDeserializer.deserialize(ReferenceTypeDeserializer.java:204)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:158)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.vanillaDeserialize(BuilderBasedDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:217)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:359)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28)
	at com.fasterxml.jackson.databind.deser.std.ReferenceTypeDeserializer.deserialize(ReferenceTypeDeserializer.java:204)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:158)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.vanillaDeserialize(BuilderBasedDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:217)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4825)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3772)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3740)
	at com.merge.api.resources.ats.candidates.CandidatesClient.list(CandidatesClient.java:121)
	at com.merge.api.resources.ats.candidates.CandidatesClient.list(CandidatesClient.java:48)
```
The distinct phone number types in the postman response were:
```
                     "phone_number_type": "OTHER"
                     "phone_number_type": "MOBILE"
```
The data used came from the `Greeenhouse - Job Boards API` sample data.

The version of the client code is here:
```
<dependency>
    <groupId>dev.merge</groupId>
    <artifactId>merge-java-client</artifactId>
    <version>1.0.8</version>
</dependency>
```
Using Client Java code:
```
        mergeClient = MergeApiClient.builder()
                .accountToken("TOKEN")
                .apiKey("KEY")
                .build();
                
        candidateList = mergeClient.ats().candidates().list(
            CandidatesListRequest.builder()
            .createdAfter(OffsetDateTime.parse("2007-12-03T10:15:30+01:00"))
            .build());
        
        
```
I was successful with this API call that returned valid data:
```
            mergeClient = MergeApiClient.builder()
                .accountToken("TOKEN")
                .apiKey("KEY")
                .build();

            candidate = mergeClient.ats().candidates().retrieve(
                    "a3632d60-21d0-46ec-96b0-23cf233e668d",
                    CandidatesRetrieveRequest.builder()
                    .includeRemoteData(true)
                    .build());

```

**Comment by dcb6** [comment]
It looks like this may be a dependency issue around your Jackson libraries

Could you share the following to help me look into this further?
1. output of `mvn dependency: tree`
2. any parts of your `pom.xml` where Jackson dependencies are defined

**Comment by stuart-alatus** [comment]
Yes, here are my Jackson versions
```
                        <dependency>
                                <groupId>com.fasterxml.jackson.core</groupId>
                                <artifactId>jackson-core</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
                        <dependency>
                                <groupId>com.fasterxml.jackson.core</groupId>
                                <artifactId>jackson-databind</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
                        <dependency>
                                <groupId>com.fasterxml.jackson.dataformat</groupId> 
                                <artifactId>jackson-dataformat-yaml</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
```

`<dep.jackson.version>2.15.4</dep.jackson.version>`

**Comment by dcb6** [comment]
Thanks! Could I get the `mvn dependency:tree` output as well?

**Comment by stuart-alatus** [comment]
I'm having trouble running that command.

**Comment by dcb6** [comment]
Could you share what the issue with the command is? We'll need this output to continue troubleshooting

**Comment by stuart-alatus** [comment]
I had an old version of `maven-dependency-plugin`. Problem fixed.

Here is part of the dependency tree.  Prefer not to post the whole dependency tree here.

```
 +- dev.merge:merge-java-client:jar:1.0.8:compile
[INFO] |  +- com.squareup.okhttp3:okhttp:jar:4.12.0:compile
[INFO] |  |  +- com.squareup.okio:okio:jar:3.6.0:compile
[INFO] |  |  |  \- com.squareup.okio:okio-jvm:jar:3.6.0:compile
[INFO] |  |  |     \- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.9.10:compile
[INFO] |  |  \- org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.8.21:compile
[INFO] |  |     +- org.jetbrains.kotlin:kotlin-stdlib:jar:1.8.21:compile
[INFO] |  |     \- org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.8.21:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.4:compile
[INFO] |  +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.12.3:compile
[INFO] |  \- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.12.3:compile
```

**Comment by stuart-alatus** [comment]
Based on the dependency tree, I updated my POM to the following:

```
<dep.jackson.version>2.15.4</dep.jackson.version>
                        <dependency>
                                <groupId>com.fasterxml.jackson.core</groupId>
                                <artifactId>jackson-core</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
                        <dependency>
                                <groupId>com.fasterxml.jackson.core</groupId>
                                <artifactId>jackson-databind</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
                        <dependency>
                                <groupId>com.fasterxml.jackson.dataformat</groupId>
                                <artifactId>jackson-dataformat-yaml</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
                        <dependency>
                                <groupId>com.fasterxml.jackson.datatype</groupId>
                                <artifactId>jackson-datatype-jdk8</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
                        <dependency>
                                <groupId>com.fasterxml.jackson.datatype</groupId>
                                <artifactId>jackson-datatype-jsr310</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
```
Which created the following dependency tree:
``
[INFO] +- dev.merge:merge-java-client:jar:1.0.8:compile
[INFO] |  +- com.squareup.okhttp3:okhttp:jar:4.12.0:compile
[INFO] |  |  +- com.squareup.okio:okio:jar:3.6.0:compile
[INFO] |  |  |  \- com.squareup.okio:okio-jvm:jar:3.6.0:compile
[INFO] |  |  |     \- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.9.10:compile
[INFO] |  |  \- org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.8.21:compile
[INFO] |  |     +- org.jetbrains.kotlin:kotlin-stdlib:jar:1.8.21:compile
[INFO] |  |     \- org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.8.21:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.4:compile
[INFO] |  +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.15.4:compile
[INFO] |  \- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.15.4:compile
```
But I still get the same error. 
`java.lang.NoSuchFieldError: READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE`

**Comment by dcb6** [comment]
Where are you running this, locally? If you haven't already ready, could you run `mvn clean` and try again?

**Comment by stuart-alatus** [comment]
I think I have solved this issue this morning. The issue was introduced in `jackson-databind:jar:2.15.x`.  There are two solutions, to back port to `jackson-databind:jar:2.14.x` or explicitly import jackson-annotations with the 2.15.x libraries.
```
                        <dependency>
                                <groupId>com.fasterxml.jackson.core</groupId>
                                <artifactId>jackson-annotations</artifactId>
                                <version>${dep.jackson.version}</version>
                        </dependency>
```

I have tested both and receive valid results for both.  I will move forward with the explicit import for now.

Let me know if you want me to try something different to confirm.

I appreciate your help with debugging this issue.
